cmake/FindCMSIS.cmake

```diff
@@ -218,9 +223,17 @@ foreach(COMP ${CMSIS_FIND_COMPONENTS_FAMILIES})
         stm32_get_chip_type(${FAMILY} ${DEVICE} TYPE)
         string(TOLOWER ${DEVICE} DEVICE_L)
         string(TOLOWER ${TYPE} TYPE_L)
+
+        set(STARTUP_NAMES startup_stm32${TYPE_L}.s startup_stm32${TYPE_L}${CORE_Ucm}.s)
+        if(${FAMILY} STREQUAL "MP1")
+            # There are stm32mp15?axx.s and stm32mp15?cxx.s files but no stm32mp15?dxx.s nor stm32mp15?fxx.s. No idea why.
+            # I think that stm32mp15xx.s should be compatible with all stm32mp15 devices anyway.
+            # This might need refinement if devices other stm32mp15 are released.
+            list(APPEND STARTUP_NAMES startup_stm32mp15xx.s)
+        endif()
```

Comment:
Original comment from @Inujel 
Regarding this:
```md
CMSIS creates the following targets:

* `CMSIS::STM32::<FAMILY>` (e.g. `CMSIS::STM32::F4`) - common includes, compiler flags and defines for family
* `CMSIS::STM32::<TYPE>` (e.g. `CMSIS::STM32::F407xx`) - common startup source for device type, depends on `CMSIS::STM32::<FAMILY>`
* `CMSIS::STM32::<DEVICE>` (e.g. `CMSIS::STM32::F407VG`) - linker script for device, depends on `CMSIS::STM32::<TYPE>`

So, if you don't need linker script, you can link only `CMSIS::STM32::<TYPE>` library and provide your own script using `stm32_add_linker_script` function
```

### About the startup file
[CMSIS provides stm32mp15?axx.s and stm32mp15?cxx.s files](https://github.com/STMicroelectronics/STM32CubeMP1/tree/1.5.0/Drivers/CMSIS/Device/ST/STM32MP1xx/Source/Templates/gcc) but no stm32mp15?dxx.s nor stm32mp15?fxx.s (I have no clue why). 
It also provides a stm32mp15xx.s file which seems to be compatible for all MP15 devices (could you give me your opinion on this?). 

I think we can simply use stm32mp15xx.s for all MP15 devices, making `CMSIS::STM32::<TYPE>` targets redundant


### About the linker script
I think that the linker script should be managed differently for the MP1 devices because instead of having a FLASH and a RAM, they have a retention RAM, and 4 RAM sections that are configurable : [STM32MP15 RAM mapping](https://wiki.st.com/stm32mpu/wiki/STM32MP15_RAM_mapping).

This is important because the developer must ensure that the RAM layout used by the linker script matches the device tree used on the A7 side.

All MP15 have the same amount of RAM.

I think that
 - There's only sense in configuring the linker per target. There's no use having `CMSIS::STM32::<DEVICE>` targets.
 - We should provide a linker script generator that takes the sizes of the RAM sections in input and support 3 use cases :
    1. Default RAM geometry:
        ```cmake
        add_executable(blinky main.cpp)
        target_link_libraries(blinky PRIVATE CMSIS::STM32::MP1::M4 CMSIS::STM32::MP1::M4::DefaultLinkerScript)
        ```
    2. Custom RAM geometry with generated linker script:
        ```cmake
        add_executable(blinky main.cpp)
        stm32_make_mp1_linker_script(BLINKY_LINKER_SCRIPT ...<ram geometry description>)
        target_link_libraries(blinky PRIVATE CMSIS::STM32::MP1 BLINKY_LINKER_SCRIPT)
        ```
    3. Custom linker script:
        ```cmake
        add_executable(blinky main.cpp)
        target_link_libraries(blinky PRIVATE CMSIS::STM32::MP1)
        stm32_add_linker_script(blinky PRIVATE my_linker_script.ld)
        ```
 
The linker script for MP1 devices is quite different from what's produced by `linker_ld.cmake`, so I think it should be easier to write another generator.

### Conclusion about targets
I don't see any point in having targets other than `CMSIS::STM32::MP1::CM4`, they will all do the exact same thing in the end.

Reply:
Alirght! Thanks for the review.